lib/dir_model/export.rb

```diff
@@ -25,18 +27,37 @@ def generated?
     def generate
       cleanup if generated?
 
-      self.class.files.each do |file_name, options|
-        dir_path  = instance_exec(&options[:path])
-        file_path = File.join(dir_path, instance_exec(&options[:name]))
+      self.class.file_names.each do |file_name|
+        options = self.class.options(file_name)
+
+        dir_path  = get_value_of(options[:path])
+        file_path = File.join(dir_path, get_value_of(options[:name]))
 
         mkdir { File.join(@root_path, dir_path) }
 
+        file_path += ".#{get_extension(file_name, options[:extensions])}"
+
         File.open(File.join(@root_path, file_path), 'wb') {|f| f.write(self.public_send(file_name).read) }
       end
     ensure
       @generated = true
     end
 
+    def get_value_of(string_or_proc)
+      return string_or_proc if string_or_proc.is_a?(String)
+      instance_exec(&string_or_proc)
+    end
+
+    def get_extension(file, extensions)
+      extension = FastImage.type(self.public_send(file))
```

Comment:
hmm.... let's simplify and just use the `filename`? it's not always images (could be a json ;)